client/text.cpp

```diff
@@ -187,8 +188,23 @@ const QString popup_info_text(struct tile *ptile, bool with_links)
     str += QString(_("Unknown"));
     return str.trimmed();
   }
-  str += QString(_("Terrain: %1")).arg(tile_get_info_text(ptile, true, 0))
-         + qbr();
+
+  {
+    struct tile_info info = tile_get_info(ptile);
+    info.terrain = maybe_link(info.terrain, HELP_TERRAIN);
+    if (!info.resource.isEmpty()) {
+      info.resource = maybe_link(info.resource, HELP_EXTRA);
+    }
+    for (auto &extra : info.extras) {
```

Comment:
Avoid an expensive detach (delayed copy) of the container:
```suggestion
    for (auto &extra : std::as_const(info.extras)) {
```
(or set the whole `tile_info` to `const` when declaring it above)

Reply:
This loop has the sole purpose of manipulating the contents of the `QStringList`. This is no longer possible when working with `const`.